library/Core.cpp

```diff
@@ -2042,6 +2040,11 @@ void Core::doUpdate(color_ostream &out, bool first_update)
     // Execute per-frame handlers
     onUpdate(out);
 
+    if (df::global::ui->main.autosave_request || (vs_changed && strict_virtual_cast<df::viewscreen_savegamest>(screen)))
+    {
+        doSave(out);
```

Comment:
Any chance we can make this only run once each time autosave_request is set? This assumes autosaves are synchronous, which won't necessarily be the case in future DF versions (see e.g. 0.40.01).

Reply:
Done in https://github.com/DFHack/dfhack/pull/1402/commits/100e584c3435edca39216550c05c5997a0ef401f